Firmware/Tupfile.lua

```diff
@@ -176,3 +148,12 @@ build{
         '.'
     }
 }
+
+if Dev_Env == "CLION" then
+    --> Rule for allowing code completion with CLion.
```

Comment:
I would prefer the CLion stuff to be a separate merge request. I cannot comment on it since I don't use it.

Reply:
Given that it is disabled by default, tup.config.default CONFIG_DEVELOPMENT_IDE=VSCODE
and that Ive extensively tested both in my dev environment, id like to ask that this remain as it is a key part of the HAL abstraction. 
The aim of this is decouple from a specific CPU, and a specific IDE such that it is as simple as adding a few extra lines to port between different processors (which may be developed in different IDEs)